In [4]:
install.packages("tidyverse")
library(tidyverse)
set.seed(42)
clinical_trial <- tibble(
    patient_ID = 1:100,
    bp_before = sample(140:180, 100, replace = TRUE) ,
    bp_after = bp_before - round(rnorm(100, mean = 15, sd = 10))
)

head(clinical_trial)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



patient_ID,bp_before,bp_after
<int>,<int>,<dbl>
1,176,166
2,140,119
3,164,141
4,149,129
5,175,169
6,157,153


In [5]:
# Performing Paired t-test
ttest_result <- t.test(clinical_trial$bp_before , clinical_trial$bp_after , paired = TRUE)
print(ttest_result)


	Paired t-test

data:  clinical_trial$bp_before and clinical_trial$bp_after
t = 15.555, df = 99, p-value < 2.2e-16
alternative hypothesis: true mean difference is not equal to 0
95 percent confidence interval:
 12.43223 16.06777
sample estimates:
mean difference 
          14.25 



* **'t = 15.555'** ==> The durg's effect is huge (far from 0).

* **'df = 99'** ==> Number of patients minus 1.

* **'p-value < 2.2e-16'** ==> Extremely small (less than 0.05); proves the drug works and it is NOT luck.

* **'95% confidence interval'** ==> Future patients will see a blood pressure drop between 12.43 and 16.06 points.

* **'mean difference = 14.25'** ==> On average, patients' blood pressure dropped by 14.25 points.